### ISSUE TITLE
Invalid generated code with latest `prance==0.21.8.0` release

### ISSUE THREAD

**Issue by mboutet** [koxudaxi/datamodel-code-generator#494]
**Describe the bug**
Latest `prance==0.21.8.0` causes invalid generated code for `datetime` field.

**To Reproduce**

Example schema:
```yaml
openapi: 3.0.3

info:
  title: Foo
  version: 0.0.1

servers: [ ]

paths:
  /foo:
    get:
      responses:
        "200":
          description: Foo
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/foo'

components:
  schemas:
    foo:
      type: object
      properties:
        created_at:
          description: Creation date
          type: string
          format: date-time
          example: 2021-04-12T23:20:50.52Z
      additionalProperties: false
      required:
        - created_at
```

Used commandline:
```
datamodel-codegen \
  --input openapi-problem.yaml \
  --output generated_models.py \
  --disable-timestamp \
  --target-python-version 3.9 \
  --strict-types str bytes int float bool \
  --validation
```

With `prance<0.21.8.0`, the generated model is:
<details>
  <summary>Click to expand!</summary>
  
```python
# generated by datamodel-codegen:
#   filename:  openapi-problem.yaml

from __future__ import annotations

from datetime import datetime

from pydantic import (
    BaseModel,
    Extra,
    Field,
)


class Foo(BaseModel):
    class Config:
        extra = Extra.forbid

    created_at: datetime = Field(..., description='Creation date', example='2021-04-12T23:20:50.52Z')
```
</details>

With `prance==0.21.8.0`, the generated model is:
<details>
  <summary>Click to expand!</summary>
  
```python
# generated by datamodel-codegen:
#   filename:  openapi-problem.yaml

from __future__ import annotations

from datetime import datetime

from pydantic import (
    BaseModel,
    Extra,
    Field,
)


class Foo(BaseModel):
    class Config:
        extra = Extra.forbid

    created_at: datetime = Field(
        ..., description='Creation date', example=datetime.datetime(2021, 4, 12, 23, 20, 50, 520000)
    )
```
</details>

**Expected behavior**

With latest prance, the `example` is `datetime.datetime(2021, 4, 12, 23, 20, 50, 520000)` but it should be `datetime(2021, 4, 12, 23, 20, 50, 520000)` or simply a string like before.

**Version:**
 - OS: macOS 11.5.1 (20G80)
 - Python version: 3.9.1
 - datamodel-code-generator version: 0.11.9

**Comment by koxudaxi** [comment]
@mboutet 
Thank you for reporting this issue.

I have understood why we get the error with prance version 0.21.8.0.
If we set a cli option ` --validation ` then `datamodel-code-generator` uses the result of prance validator as a YAML source.
https://github.com/koxudaxi/datamodel-code-generator/blob/b782b5df6de4355070420d05ce88de15f6c21602/datamodel_code_generator/parser/openapi.py#L340-L348

prance uses runamel.yml for YAML parser since version 0.21.8.0 
https://github.com/RonnyPfannschmidt/prance/commit/dbff665a571715b2f7d40717e3ea95876375a88b
The changes affect the behavior and you got the invalid generated code.

I just created a PR to fix the bug.
https://github.com/koxudaxi/datamodel-code-generator/pull/496/files
In PR, the method doesn't use the result of prance. It means the method parses YAML two times.
That is increased the run time.
I guess the impact is very small on the time in almost case.

**Comment by mboutet** [comment]
Thanks a lot!

**Comment by koxudaxi** [comment]
@mboutet 
I have released a new version as `0.11.11` :rocket:

**Comment by mboutet** [comment]
Works perfectly, thanks a lot!
